### PR TITLE
disttask: fix data race in the dispatcher test

### LINKED ISSUE #42369: DATA RACE in the dispatcher_test.TestSimple
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 3. What did you see instead (Required)

```
==================
WARNING: DATA RACE
Write at 0x00c01c8522d0 by goroutine 54663:
  runtime.mapassign_fast64()
      GOROOT/src/runtime/map_fast64.go:93 +0x0
  github.com/pingcap/tidb/disttask/framework/dispatcher.(*dispatcher).setRunningGlobalTasks()
      disttask/framework/dispatcher/dispatcher.go:62 +0x170
  github.com/pingcap/tidb/disttask/framework/dispatcher.(*dispatcher).DispatchTaskLoop()
      disttask/framework/dispatcher/dispatcher.go:143 +0xa51
  github.com/pingcap/tidb/disttask/framework/dispatcher.(*dispatcher).DispatchTaskLoop-fm()
      <autogenerated>:1 +0x39
  github.com/pingcap/tidb/util.(*WaitGroupWrapper).Run.func1()
      util/wait_group_wrapper.go:154 +0x73
Previous read at 0x00c01c8522d0 by goroutine 27062:
  github.com/pingcap/tidb/disttask/framework/dispatcher_test.TestSimple()
      disttask/framework/dispatcher/dispatcher_test.go:155 +0x1c9
  github.com/pingcap/tidb/session.BootstrapSession()
      session/session.go:3314 +0x6d3
  github.com/pingcap/tidb/domain.(*Domain).GetSessionCache()
      domain/sysvar_cache.go:62 +0x5c
  github.com/pingcap/tidb/session.(*session).loadCommonGlobalVariablesIfNeeded()
      session/session.go:3656 +0x104
  github.com/pingcap/tidb/session.(*session).ExecuteStmt()
      session/session.go:2093 +0x145
  github.com/pingcap/tidb/session.(*session).ExecuteInternal()
      session/session.go:1628 +0x31b
  github.com/pingcap/tidb/domain.(*Domain).LoadPrivilegeLoop()
      domain/domain.go:1414 +0x130
  github.com/pingcap/tidb/session.BootstrapSession()
      session/session.go:3307 +0x684
  github.com/pingcap/tidb/testkit.bootstrap()
      testkit/mockstore.go:85 +0xac
  github.com/pingcap/tidb/testkit.CreateMockStoreAndDomain()
      testkit/mockstore.go:70 +0xe9
  github.com/pingcap/tidb/testkit.CreateMockStore()
      testkit/mockstore.go:62 +0x3b1
  github.com/pingcap/tidb/disttask/framework/dispatcher_test.MockDispatcher()
      disttask/framework/dispatcher/dispatcher_test.go:37 +0x55
  github.com/pingcap/tidb/disttask/framework/dispatcher_test.TestSimple()
      disttask/framework/dispatcher/dispatcher_test.go:142 +0x4a
  testing.tRunner()
      GOROOT/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      GOROOT/src/testing/testing.go:1629 +0x47
Goroutine 54663 (running) created at:
  github.com/pingcap/tidb/util.(*WaitGroupWrapper).Run()
      util/wait_group_wrapper.go:152 +0xe4
  github.com/pingcap/tidb/disttask/framework/dispatcher.(*dispatcher).Start()
      disttask/framework/dispatcher/dispatcher.go:98 +0x9c
  github.com/pingcap/tidb/disttask/framework/dispatcher_test.TestSimple()
      disttask/framework/dispatcher/dispatcher_test.go:143 +0x73
  github.com/pingcap/tidb/session.BootstrapSession()
      session/session.go:3314 +0x6d3
  github.com/pingcap/tidb/domain.(*Domain).GetSessionCache()
      domain/sysvar_cache.go:62 +0x5c
  github.com/pingcap/tidb/session.(*session).loadCommonGlobalVariablesIfNeeded()
      session/session.go:3656 +0x104
  github.com/pingcap/tidb/session.(*session).ExecuteStmt()
      session/session.go:2093 +0x145
  github.com/pingcap/tidb/session.(*session).ExecuteInternal()
      session/session.go:1628 +0x31b
  github.com/pingcap/tidb/domain.(*Domain).LoadPrivilegeLoop()
      domain/domain.go:1414 +0x130
  github.com/pingcap/tidb/session.BootstrapSession()
      session/session.go:3307 +0x684
  github.com/pingcap/tidb/testkit.bootstrap()
      testkit/mockstore.go:85 +0xac
  github.com/pingcap/tidb/testkit.CreateMockStoreAndDomain()
      testkit/mockstore.go:70 +0xe9
  github.com/pingcap/tidb/testkit.CreateMockStore()
      testkit/mockstore.go:62 +0x3b1
  github.com/pingcap/tidb/disttask/framework/dispatcher_test.MockDispatcher()
      disttask/framework/dispatcher/dispatcher_test.go:37 +0x55
  github.com/pingcap/tidb/disttask/framework/dispatcher_test.TestSimple()
      disttask/framework/dispatcher/dispatcher_test.go:142 +0x4a
  testing.tRunner()
      GOROOT/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      GOROOT/src/testing/testing.go:1629 +0x47
Goroutine 27062 (running) created at:
  testing.(*T).Run()
      GOROOT/src/testing/testing.go:1629 +0x805
  testing.runTests.func1()
      GOROOT/src/testing/testing.go:2036 +0x8d
  testing.tRunner()
      GOROOT/src/testing/testing.go:1576 +0x216
  testing.runTests()
      GOROOT/src/testing/testing.go:2034 +0x87c
  testing.(*M).Run()
      GOROOT/src/testing/testing.go:1906 +0xb44
  go.uber.org/goleak.VerifyTestMain()
      external/org_uber_go_goleak/testmain.go:53 +0x70
  github.com/pingcap/tidb/disttask/framework/dispatcher.TestMain()
      disttask/framework/dispatcher/main_test.go:44 +0x544
  main.main()
      bazel-out/k8-fastbuild/bin/disttask/framework/dispatcher/dispatcher_test_/testmain.go:104 +0x5e8
================== 
```

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #42369

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: disttask/framework/dispatcher/dispatcher_test.go [hunk 1]
```diff
@@ -151,9 +151,8 @@ func TestSimple(t *testing.T) {
 	cnt := 40
 	taskMap := make(map[int64]*proto.Task)
 	for i := 0; i < cnt; i++ {
-		taskMap = dsp.(dispatcher.DispatcherForTest).GetRunningGlobalTasks()
-		if len(taskMap) == 1 {
-			break
+		if dsp.(dispatcher.DispatcherForTest).RunningGlobalTasksCnt() == 1 {
+			taskMap = dsp.(dispatcher.DispatcherForTest).GetRunningGlobalTasks()
```
Comment (zimulala) [inline disttask/framework/dispatcher/dispatcher_test.go:155]: I think there will also be a data race in line159 or line165.
The current code does not support parallelism. I will fix https://github.com/pingcap/tidb/issues/42383. Or do you want to change it?

Comment (hawkingrei) [reply disttask/framework/dispatcher/dispatcher_test.go:155]: If you fix it in your PR, you can try to enable the race for verifying it.

Comment (hawkingrei) [reply disttask/framework/dispatcher/dispatcher_test.go:155]: Please close the issue after your pr merging.

Comment (zimulala) [reply disttask/framework/dispatcher/dispatcher_test.go:155]: OK.

**Pull request closed without merging**